components/eam/src/physics/cam/micro_p3_interface.F90

```diff
@@ -847,13 +814,12 @@ subroutine get_prescribed_CCN(nccn_prescribed,micro_p3_lookup_dir,its,ite,kts,kt
 
       !interpolate between mon_ccn_1 and mon_ccn_2 to calculate nccn_prescribed
       !based on current date
-      call pbuf_get_field(pbuf,mon_ccn_idx(month), mon_ccn_1)
 
-      call pbuf_get_field(pbuf,mon_ccn_idx(next_month), mon_ccn_2)
+      call pbuf_get_field(pbuf,mon_ccn_idx, mon_ccn)
 
-      fraction_of_month = (day*3600.0*24.0 + tod)/(3600*24*days_per_month(current_month)) !tod is in seconds
+      fraction_of_month = (day*3600.0*24.0 + tod)/(3600*24*days_per_month(month)) !tod is in seconds
 
-      nccn_prescribed = mon_ccn_1*(1-fraction_of_month) + mon_ccn_2*(fraction_of_month)
+      nccn_prescribed = mon_ccn(:,:,month)*(1-fraction_of_month) + mon_ccn(:,:,next_month)*(fraction_of_month)
```

Comment:
So here it looks like we are loading the full range of CCN values from pbuf and then just using `month` and `next_month` for the interpolation.  But in the radiation code changes we are still using the old approach.  Would it make sense (or is it even possible) to make these consistent?

Reply:
Oh nevermind, it looks like you are using the same approach.